docs/lab/05/index.md

```diff
@@ -0,0 +1,209 @@
+
+:::info
+Asynchronous programming is widely used in web development. In JavaScript, the equivalent of a `Future` would be a `Promise`.
+:::
+
+Read more about how async/await works in Rust [here](https://rust-lang.github.io/async-book/01_getting_started/01_chapter.html).
+
+## Channels
+
+Up to this point, to be able to share peripherals across multiple tasks, we have been using global `Mutex`s or passing them directly as parameters to the tasks. But there are other, more convenient ways to send data to and from tasks. Instead of having to make global, static variables that are shared by tasks, we could choose to only send the information that we need from one task to another. To achieve this, we can use *channels*.
+
+**Channels** allow a unidirectional flow of information between two endpoints: the *Sender* and the *Receiver*. The sender sends a value through the channel, and the receiver receives this value once it is ready to do so. Until it is ready, the data will be stored inside a queue. Channels in Embassy are *Multiple Producer, Multiple Consumer*, which means that we can have a channel associated with multiple senders and multiple receivers. 
+
+:::info
+Executors actually internally use channels when dealing with tasks. (TODO: how?)
+:::
+
+To use a channel in Embassy, we first need to declare a static instance of the channel. 
+
+```rust
+static CHANNEL: Channel<ThreadModeRawMutex, bool, 64> = Channel::new();
+```
+
+The second generic argument of `Channel` is the type of data that we will be sending through the channel. The third argument is the maximum number of values that can be stored in the queue. 
+
+Let's say we spawn a task `task1` that runs a timer. Every second, we want to toggle an LED in the `main` function, based on the timer running in `task1`. For this, `task1` would need to 
+send a signal to the `main` program every time the 1 second alarm has fired, meaning the task and the main program would share the channel. `task1` would *send* over the channel, and 
+`main` would *receive*.
+
+For this we need to pass a `Sender` endpoint to `task1` once we spawn it.
+
+```rust
+// ---- fn main() ----
+spawner.spawn(task1(CHANNEL.sender())).unwrap();
+```
+
+The `sender()` function returns a new sender for the given channel.
+
+Inside `task1`, we would just set a timer and wait until it fires. After it fires, we send a signal through the channel, to indicate that 1 second has elapsed.
+
+```rust
+#[embassy_executor::task]
+async fn task1(channel_sender: Sender<'static, ThreadModeRawMutex, bool, 64>) {
+    loop {
+        Timer::after_secs(1).await;
+        channel_sender.send(true).await;
+    }
+}
+```
+
+In the `main` function, we need to then wait for the signal, and once it's received, toggle the LED.
+
+```rust
+// ---- fn main() ----
+loop {
+    let value = CHANNEL.receive().await;
+    match value {
+        true => led.toggle().unwrap(),
+        false => info!("We got something else")
+    }
+}
+```
+
+:::info
```

Comment:
Add a link to the Rust book where it explains ownership and borrowing.

Reply:
Done.